sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StorageAllowedHeadersAndQueries.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.implementation;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public final class StorageAllowedHeadersAndQueries {
+    public static final class BlobHeadersAndQueries {
+        private static final Set<String> BLOB_HEADERS = Stream.of(
+            "Access-Control-Allow-Origin", "Cache-Control", "Content-Length", "Content-Type", "Date", "Request-Id",
+            "traceparent", "Transfer-Encoding", "User-Agent", "x-ms-client-request-id", "x-ms-date", "x-ms-error-code",
+            "x-ms-request-id", "x-ms-return-client-request-id", "x-ms-version", "Accept-Ranges", "Content-Disposition",
+            "Content-Encoding", "Content-Language", "Content-MD5", "Content-Range", "ETag", "Last-Modified", "Server",
+            "Vary", "x-ms-content-crc64", "x-ms-copy-action", "x-ms-copy-completion-time", "x-ms-copy-id",
+            "x-ms-copy-progress", "x-ms-copy-status", "x-ms-has-immutability-policy", "x-ms-has-legal-hold",
+            "x-ms-lease-state", "x-ms-lease-status", "x-ms-range", "x-ms-request-server-encrypted",
+            "x-ms-server-encrypted", "x-ms-snapshot", "x-ms-source-range", "If-Match", "If-Modified-Since",
+            "If-None-Match", "If-Unmodified-Since", "x-ms-access-tier", "x-ms-access-tier-change-time",
```

Comment:
What about metadata headers, anything prefixed with "x-ms-meta" and any other user defined headers? For example, what if they set custom headers to give information to a proxy? It looks like there's no way to log those and they'll always get redacted.

Reply:
Correct, this is a problem across all languages right now.  Users would have to add specific metadata headers they wanted logged.  I don't think we'll be able to add support for that before the next release, but I'm confident we could figure out a way to do that without breaking in the future.

Reply:
@rickle-msft I have changed my implementation which customer is able to remove, add headers on the top of the core and storage list.